great_expectations/datasource/fluent/sources.py

```diff
@@ -386,7 +387,12 @@ def update_datasource(name: str, **kwargs) -> Datasource:
             return self.create_add_crud_method(datasource_type)(name=name, **kwargs)
 
         update_datasource.__doc__ = doc_string
-        # TODO: update signature
+        update_datasource.__signature__ = _merge_signatures(
```

Comment:
@Kilo59 I ❤️ the `_merge_signatures()` method -- would it make sense to add a test for something so core as this (does not need to be in the present scope).  Thanks.

Reply:
We test it indirectly in `test_metadatasource.py`